### PR TITLE
[NVPTX] Add folding for cvt.rn.bf16x2.f32

### PR DESCRIPTION
**Pull request by AlexMaclean**

### CODE REVIEW

File: llvm/test/CodeGen/NVPTX/convert-sm80.ll [hunk 1]
```diff
@@ -135,3 +135,24 @@ ret i32 %val
 }
 
 declare i32 @llvm.nvvm.f2tf32.rna(float)
+
+
+define <2 x bfloat> @fold_ff2bf16x2(float %a, float %b) {
+; CHECK-LABEL: fold_ff2bf16x2
```
Comment (Artem-B) [inline llvm/test/CodeGen/NVPTX/convert-sm80.ll:141]: We may as well automate check generation in this test file, too. For vector construction, theorder of inputs is important and we're not capturing it in this file right now.

**Review by Artem-B — APPROVED** [review]
Nice. LGTM modulo test nit.

File: llvm/test/CodeGen/NVPTX/convert-sm80.ll (hunk 1 above)
Comment (AlexMaclean) [reply llvm/test/CodeGen/NVPTX/convert-sm80.ll]: Sounds good, I've updated the test file.

**Comment by tomnatan30** [comment]
This change is breaking triton tests (results in huge numeric disparities, e.g. https://github.com/triton-lang/triton/blob/main/python/test/unit/language/test_core.py), we'll need to revert until a fix forward can be merged.

**Comment by justinfargnoli** [comment]
Thank you for identifying and reverting this @tomnatan30!

**Pull request merged**